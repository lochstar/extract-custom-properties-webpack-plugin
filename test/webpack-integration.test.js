import fs from 'fs'
import path from 'path'
import webpack from 'webpack'
import { readFileOrEmpty, checkForWebpackErrors } from './util/helpers'

const cases = process.env.CASES ? process.env.CASES.split(',') : fs.readdirSync(path.join(__dirname, 'cases'))

describe('Webpack Integration Tests', () => {
  cases.forEach((testCase) => {
    if (/^_skip_/.test(testCase)) return;
    it(testCase, (done) => {
      const testDirectory = path.join(__dirname, 'cases', testCase)
      const outputDirectory = path.join(__dirname, 'js', testCase)
      const expectedDirectory = path.join(testDirectory, 'expected')

      const configFile = path.join(testDirectory, 'webpack.config.js')
      const config = Object.assign(
        fs.existsSync(configFile) ? require(configFile) : { entry: { test: './index.js' } },
        {
          mode: 'production',
          context: testDirectory,
          output: {
            filename: '[name].js',
            path: outputDirectory
          }
        }
      )

      webpack(config, (err, stats) => {
        checkForWebpackErrors({ err, stats, done })
        fs.readdirSync(expectedDirectory).forEach((file) => {
          const expectedFile = readFileOrEmpty(path.join(expectedDirectory, file))
          const actualFile = readFileOrEmpty(path.join(outputDirectory, file))
          expect(actualFile).toEqual(expectedFile)
          expect(actualFile).toMatchSnapshot()
        })
        done()
      })
    })
  })
})
