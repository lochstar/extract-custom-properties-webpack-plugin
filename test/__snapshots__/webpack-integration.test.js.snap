// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack Integration Tests assetNameRegExp 1`] = `
"{
  \\"dashedColor\\": {
    \\"color\\": [
      \\"a\\"
    ],
    \\"background-color\\": [
      \\".class2:after\\"
    ]
  },
  \\"dividerColor\\": {
    \\"border-color\\": [
      \\"code\\"
    ],
    \\"color\\": [
      \\".class3\\"
    ]
  },
  \\"baseColor\\": {
    \\"color\\": [
      \\".class1\\"
    ],
    \\"border-color\\": [
      \\".class1\\"
    ]
  },
  \\"sizeH1\\": {
    \\"font-size\\": [
      \\".class1\\"
    ]
  },
  \\"camelColor\\": {
    \\"color\\": [
      \\".class2\\",
      \\".class4:not(:first-child)\\",
      \\".classCamel\\"
    ]
  }
}"
`;

exports[`Webpack Integration Tests default 1`] = `
"{
  \\"dashedColor\\": {
    \\"color\\": [
      \\"a\\"
    ],
    \\"background-color\\": [
      \\".class2:after\\"
    ]
  },
  \\"dividerColor\\": {
    \\"border-color\\": [
      \\"code\\"
    ],
    \\"color\\": [
      \\".class3\\"
    ]
  },
  \\"baseColor\\": {
    \\"color\\": [
      \\".class1\\"
    ],
    \\"border-color\\": [
      \\".class1\\"
    ]
  },
  \\"sizeH1\\": {
    \\"font-size\\": [
      \\".class1\\"
    ]
  },
  \\"camelColor\\": {
    \\"color\\": [
      \\".class2\\",
      \\".class4:not(:first-child)\\",
      \\".classCamel\\"
    ]
  }
}"
`;

exports[`Webpack Integration Tests filename 1`] = `
"{
  \\"dashedColor\\": {
    \\"color\\": [
      \\"a\\"
    ],
    \\"background-color\\": [
      \\".class2:after\\"
    ]
  },
  \\"dividerColor\\": {
    \\"border-color\\": [
      \\"code\\"
    ],
    \\"color\\": [
      \\".class3\\"
    ]
  },
  \\"baseColor\\": {
    \\"color\\": [
      \\".class1\\"
    ],
    \\"border-color\\": [
      \\".class1\\"
    ]
  },
  \\"sizeH1\\": {
    \\"font-size\\": [
      \\".class1\\"
    ]
  },
  \\"camelColor\\": {
    \\"color\\": [
      \\".class2\\",
      \\".class4:not(:first-child)\\",
      \\".classCamel\\"
    ]
  }
}"
`;
